timemachine/cpp/src/centroid_restraint.cu

```diff
@@ -65,12 +61,9 @@ void CentroidRestraint<RealType>::execute_device(
             kb_,
             b0_,
             d_du_dx,
-            d_u == nullptr ? nullptr : d_u_buffer_);
+            d_u // Can write directly to the energy buffer for this potential.
```

Comment:
I think this is pretty dangerous - the input API assumes that `__int128 *d_u` can set to some initial value `x`, and this function can increment it by `y` resulting in `x+y` (like the forces). However, the actual internal kernel call directly *sets* `d_u` via:
``` cpp
//  k_centroid_restraint.cuh
if (t_idx == 0 && d_u) {
    RealType nrg = kb * (dij - b0) * (dij - b0);
    d_u[t_idx] = FLOAT_TO_FIXED_ENERGY<RealType>(nrg);
}
```
i.e. it blows away the old value  of `x` - we should talk about the semantics of `execute_device` more carefully tmrw.